### ISSUE TITLE
Case sensitivity and name uniqueness

### ISSUE THREAD

**Issue by joesiltberg** [dotse/tls-fed-auth#2]
The "entity_id" attribute for an entity, as well as the "name" attribute for servers and clients should perhaps be specified in more in detail.

For instance, should they be considered case-sensitive or not?

The spec says the entity_id is "usually a domain name", but I suppose it can be any string? Can it contain whitespace for instance? Perhaps an implementation wants to store the cert on disk, can the entity_id be used as part of the path? Could it be a part of a URL or would it need to be URL-encoded?

Possibly we should even have a note about sanitizing the strings before using them in for instance SQL queries :-)

Another related issue is uniqueness. entity_id's are unique within the metadata (should this be stated explicitly?), but what about server and client names? Must they be unique within the whole metadata or just unique within the entity (I assume the latter so a server can be named something generic such as "Test" or "Production").

**Comment by jschlyter** [comment]
We'll change `entity_id` to URI and all matching rules are given, see fe1732f3794afe9114ac3a5b2c68ca4cf0dda847

**Comment by jschlyter** [comment]
entity_id uniqueness must be ensured by the federation operator.

**Comment by joesiltberg** [comment]
URIs make it well defined in the spec, but difficult to implement correctly (comparing URIs correctly is non-trivial). Maybe just specify that it must be a domain name? That should still give us all the rules?

I noticed the server name attribute was replaced by a description. Up until now, the user of the EGIL client specifies which server to communicate with by supplying an entity_id and the name of the server. This combination is also what I would have expected to store in a configuration of a scheduled recurring job. But after this change I'm wondering what should be used instead. base_uri for the server? I avoided using that before, since I thought more or less temporary URIs could be used here (like the hostname of an AWS EC2 server). 

I'm not sure if I'm explaining the problem properly, perhaps it's easier to give some use cases/requirements to think around:

1. When running the command line client, we need to be able to uniquely identify a server in the command line arguments.

2. The way to uniquely identify the server above should be reasonable to store in a configuration file for a recurring job (so more or less permanent as long as the server still exists). 

3. What's used by the command line, or in configuration files, needn't be user friendly. But it should be possible to create a user friendly interface for choosing a server.

So maybe the current state with a base_uri and description is fine, as long as it's clear that base_uri's listed in the metadata should be long living URIs. Alternatively bring back the name attribute (and keep description).
